core/services/aws/bloodDonation/acceptDonationRequest.ts

```diff
@@ -60,21 +67,42 @@ async function acceptDonationRequest(
       AcceptDonationRequestModel
       >(new AcceptDonationRequestModel())
     )
-    if (acceptanceRecord !== null && acceptanceRecord.status === AcceptDonationStatus.COMPLETED) {
+    if (isAlreadyDonated(acceptanceRecord)) {
       throw new Error('You already donated.')
     }
 
-    if ((status === AcceptDonationStatus.ACCEPTED && acceptanceRecord !== null)) {
+    if (acceptanceRecord !== null && status === acceptanceRecord.status) {
       return generateApiGatewayResponse(
-        { message: 'Donation request accepted successfully.' },
+        { message: 'Donation request already accepted successfully.' },
         HTTP_CODES.OK
       )
     }
 
-    if (status === AcceptDonationStatus.ACCEPTED && acceptanceRecord === null) {
-      await acceptBloodDonationRequest(donorId, seekerId, requestPostId, createdAt, status)
+    const userProfile = await userService.getUser(
+      donorId,
+      new DynamoDbTableOperations<UserDetailsDTO, UserFields, UserModel>(new UserModel())
+    )
+    const donationPost = await getDonationRequest(seekerId, requestPostId, createdAt)
+
+    if (userProfile.bloodGroup !== donationPost.requestedBloodGroup) {
+      throw new Error('Your blood group doesn\'t match with the request blood group')
+    }
+
+    if (acceptanceRecord === null) {
+      if (status === AcceptDonationStatus.ACCEPTED) {
+        await createAcceptanceRecord(donorId, seekerId, createdAt, requestPostId, userProfile)
+        await sendNotificationToSeeker(
+          seekerId,
+          requestPostId,
+          donationPost,
+          donorId,
+          createdAt,
+          status,
+          userProfile
+        )
+      }
     } else {
-      if (acceptanceRecord !== null) {
+      if (status === AcceptDonationStatus.IGNORED) {
```

Comment:
This logic feels a little confusing,
you're saying that the acceptanceRecord can be null only when status === Accepted ?

Reply:
no, here the logic is
if acceptanceRecord !== null and then if status == ignore then we should delete the acceptance record